app/models/payment_provider_customers/base_customer.rb

```diff
@@ -10,6 +10,8 @@ class BaseCustomer < ApplicationRecord
     has_many :payments
     has_many :refunds, foreign_key: :payment_provider_customer_id
 
+    validates :customer_id, uniqueness: { scope: :type }
```

Comment:
Rails/UniqueValidationWithoutIndex: Uniqueness validation should have a unique index on the database column.